Formula/stepci.rb

```diff
@@ -0,0 +1,24 @@
+require "language/node"
+
+class Stepci < Formula
+  desc "API Testing and Monitoring made simple"
+  homepage "https://stepci.com"
+  url "https://registry.npmjs.org/stepci/-/stepci-2.1.0.tgz"
+  sha256 "bcc265995c0539944c9739fb192874326bceba9a4d928111455880538620c72a"
+  license "MPL-2.0"
+
+  livecheck do
+    url :stable
+  end
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert true
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I'm the author of Step CI, thank you so much for a detailed answer!
One question I have is: our software depends on a configuration file. Would it be appropriate to create a (temporary) file from the test method?

Let me know. Thanks

Reply:
We should be aware, that running a test against e.g. `www.example.com` *could* fail (site being down, proxy blocking access, etc). We'd produce a false positive here.
I know that this might feel like a very rare edge case. But trust me, it happens and might confuse users.
Source: I'm behind a very restrictive proxy that even breaks up ssl encryption. :(

Reply:
We could send a test request to a domain that never resolves (like brew.stepci.com). Then we can validate that the workflow resulted with an error

Reply:
Or just a Homebrew domain, we'll want to make sure the test doesn't fail then (and we optimise tests for our CI, not for local proxies)